minimint-api/src/db/mod.rs

```diff
@@ -45,7 +45,41 @@ pub trait Database: Send + Sync {
 
     fn raw_find_by_prefix(&self, key_prefix: &[u8]) -> PrefixIter<'_>;
 
-    fn raw_apply_batch(&self, batch: DbBatch) -> Result<()>;
+    fn transaction<'a>(&'a self) -> Result<Box<dyn Transaction<'a>>>;
```

Comment:
I'm not a fan of nouns for methods. Using a noun suggests a simple getter. `start_transaction`?

Reply:
I don't think we can pass`X<'a>` around though a dynamic polymorphism boundaries of modules...

Reply:
Never mind. I was wrong. It was just `#[async_trait]` requiring not to skip `<'_>` in traits.

Please carry on.